### PR TITLE
Update boardwalk install process (closes #21)

### LINKED ISSUE #21: Update Boardwalk install process
- [x] ~~Add repositories necessary to install Boardwalk in development mode as submodules~~ Submodules pin commits so this isn't a good idea anymore
- [x] Automatically clone repositories in install_boostrap if being installed in development mode
- [x] Update repository links to point to new DataBiosphere repos
- [x] Update documentation to reflect the above

┆Issue is synchronized with this [JIRA Task](https://ucsc-cgl.atlassian.net/browse/DEP-9)
┆Project Name: CGP-Deployment
┆Issue Number: DEP-9

### PR DESCRIPTION
**Pull request by natanlao**

### CODE REVIEW

File: boardwalk/README.md [hunk 1]
```diff
@@ -37,15 +37,9 @@ Here is a summary of what you need to do in order to create a Google OAuth2 app:
 
 The boardwalk installer has the option to install boardwalk and its components in either devevelopment or production mode (dev/prod). If you are installing in production, skip this section and head to **Installation Questions**. Otherwise, keep reading.
 
-To make deployment during development faster, dev mode will assume you have cloned *dcc-dashboard, dcc-dashboard-service, and dcc-metadata-indexer* under *dcc-ops/boardwalk* If you are running dcc-ops in dev mode and you haven't already, clone the repos by running the following from within `dcc-ops/boardwalk`:
+To make deployment during development faster, the `install_bootstrap` script will clone [cgp-dashboard](https://github.com/DataBiosphere/cgp-dashboard), [cgp-dashboard-service](https://github.com/DataBiosphere/cgp-dashboard-service), and [dcc-metadata-indexer](https://github.com/BD2KGenomics/dcc-metadata-indexer) under `cgp-deployment/boardwalk/` if it looks like you haven't already done so. (If you're working off a non-default branch in one of those repos, you're going to want to do this yourself.)
```
Comment (coverbeck) [inline boardwalk/README.md]: We don't use dcc-metadata-indexer in feature/commons branch. Looks like I hadn't cleaned removed the reference to it.

We will need to clone DataBiosphere/boardwalk

File: install_bootstrap [hunk 2]
```diff
@@ -335,8 +342,14 @@ CONFIG
       mustache boardwalk_launcher_config/boardwalk.config boardwalk/conf/boardwalk.config.template > boardwalk/.env
 
       cd boardwalk
-      #Bringing stuff down in case there are some cached containers
-      echo "Bringing down any existing Boardwalk container and volumes"
+      # If in dev mode, check for dev repos and clone if not present
+      if [ "${boardwalk_mode}" = "dev" ]; then
+        check_for_repo https://github.com/DataBiosphere/cgp-dashboard.git dcc-dashboard
+        check_for_repo https://github.com/DataBiosphere/cgp-dashboard-service.git dcc-dashboard-service
+        check_for_repo https://github.com/BD2KGenomics/dcc-metadata-indexer.git dcc-metadata-indexer
```
Comment (coverbeck) [inline install_bootstrap]: See previous comment about metadata indexer and boardwalk.

**Review by coverbeck — CHANGES_REQUESTED** [review]

**Comment by coverbeck** [comment]
These changes look good, but, I just noticed there are a bunch of other things in the README that need to be fixed as well. For example, references to metadata-indexer, instructions to type random strings...

Do you want to keep this PR scoped to this change and handle the rest in other PRs, or do you want to try to get it all in good?

**Comment by natanlao** [comment]
If it's all the same to you, I'll keep this PR scoped and submit another PR
in the coming days?

**Review by coverbeck — APPROVED** [review]
Agreed, tackle the other points in another PR.

LGTM.

**Pull request merged**